### PR TITLE
Remove formula tmp dir

### PR DESCRIPTION
**Pull request by kaduartur**
### Description
The need to copy the formula to a temporary folder has been removed, thus reducing the necessary IO when executing a formula.

### How to verify it
This improvement should not change the functionality of the rit.

### Changelog
Performance - remove tmp dir to formulas

### CODE REVIEW

**Review by lucasdittrichzup — APPROVED** [review]

**Comment by henriquemoraeszup** [comment]
can you make an issue and link to your project?

File: pkg/formula/runner/docker/runner_test.go [hunk 1]
```diff
@@ -184,13 +165,20 @@ func TestRun(t *testing.T) {
...
+			docker := NewRunner(in.inputResolver, in.preRun, in.fileManager, in.env, homeDir)
 			got := docker.Run(in.def, api.Prompt, false, nil)
 
-			if tt.out.err != nil && got != nil && tt.out.err.Error() != got.Error() {
-				t.Errorf("Run(%s) got %v, want %v", tt.name, got, tt.out.err)
+			if got != nil || tt.out.err != nil {
+				assert.EqualError(t, tt.out.err, got.Error())
 			}
+
+			_ = os.Chdir(pwd)
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/docker/runner_test.go:181]: Can we test for file creation on the correct dir? This is the purpose of this PR correct?

File: pkg/formula/runner/local/runner_test.go [hunk 2]
```diff
@@ -69,118 +72,90 @@ func TestRun(t *testing.T) {
...
+
 			in := tt.in
-			local := NewRunner(in.postRun, in.inputResolver, in.preRun, in.fileManager, in.env, homeDir)
+			local := NewRunner(in.inputResolver, in.preRun, in.fileManager, in.env, homeDir)
 			got := local.Run(in.def, api.Prompt, false, nil)
 
-			if tt.out.err != nil && got != nil && tt.out.err.Error() != got.Error() {
-				t.Errorf("Run(%s) got %v, want %v", tt.name, got, tt.out.err)
+			if tt.want != nil || got != nil {
+				assert.EqualError(t, got, tt.want.Error())
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/local/runner_test.go:155]: Same here

**Review by henriquemoraeszup — CHANGES_REQUESTED** [review]
Just requesting since it was already approved

File: pkg/formula/runner/docker/runner_test.go (hunk 1 above)
Comment (kaduartur) [reply pkg/formula/runner/docker/runner_test.go:181]: Which file? this PR just stops copying the `/bin` folder of a formula to a tmp folder.

**Review by victor-schumacher — DISMISSED** [review]

**Review by brunasilvazup — DISMISSED** [review]

**Review by henriquemoraeszup — APPROVED** [review]
🎖️

**Review by henriquemoraeszup — CHANGES_REQUESTED** [review]
The `pwd` of the formulas are correct, but if you create a formula with code that creates a file, on the first run the file will be created on the `.rit` folder. Subsequent runs creates on the user `pwd`

The formula has to create a file **before** its first build

**Comment by henriquemoraeszup** [comment]
It would be interesting to simulate an execution that creates a file and checks its creation on the correct folder

**Comment by kaduartur** [comment]
> The `pwd` of the formulas are correct, but if you create a formula with code that creates a file, on the first run the file will be created on the `.rit` folder. Subsequent runs creates on the user `pwd`
> 
> The formula has to create a file **before** its first build

I fixed this bug in [this](https://github.com/ZupIT/ritchie-cli/pull/875/commits/2259b48ed152dfda34c44f8805f573e66e7ebac0) commit

File: pkg/formula/runner/docker/runner_test.go [hunk 3]
```diff
@@ -184,13 +165,23 @@ func TestRun(t *testing.T) {
...
+			docker := NewRunner(in.inputResolver, in.preRun, in.fileManager, in.env, homeDir)
 			got := docker.Run(in.def, api.Prompt, false, nil)
 
-			if tt.out.err != nil && got != nil && tt.out.err.Error() != got.Error() {
-				t.Errorf("Run(%s) got %v, want %v", tt.name, got, tt.out.err)
+			if got != nil || tt.out.err != nil {
+				assert.EqualError(t, tt.out.err, got.Error())
+			} else {
+				fileCreted := filepath.Join(formulaPath, "test.txt")
+				assert.FileExists(t, fileCreted)
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/docker/runner_test.go:180]: * fileCreated

File: pkg/formula/runner/local/pre_run_test.go [hunk 4]
```diff
@@ -153,86 +153,6 @@ func TestPreRun(t *testing.T) {
 				err:     errors.New("invalid character 'e' looking for beginning of value"),
 			},
 		},
-		{
-			name: "create work dir error",
```
Comment (henriquemoraeszup) [inline pkg/formula/runner/local/pre_run_test.go:157]: code deletion ❤️

**Review by brunasilvazup — APPROVED** [review]

**Comment by henriquemoraeszup** [comment]
The implementation is breaking on Windows, just create a formula and run to see

**Review by henriquemoraeszup — APPROVED** [review]

**Pull request merged**